### ISSUE TITLE
Enhancement: [no-unnecessary-condition] detect unnecessary calls to Array.isArray

### ISSUE THREAD

**Issue by ulrichstark** [typescript-eslint/typescript-eslint#11716]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-unnecessary-condition

### Description

I constantly see LLMs generating code that checks variables by calling Array.isArray on them despite them being strictly typed as array. Adding logic into `no-unnecessary-condition` to detect these redundant calls would reduce code by avoiding unnecessary conditions and helps in reviewing AI generated code.

### Fail

```TypeScript
const items: number[] = [1, 2, 3];
if (Array.isArray(items)) {
  console.log("items is an array");
}
```

### Pass

```TypeScript
const items: number[] | null = [1, 2, 3];
if (Array.isArray(items)) {
  console.log("items is an array");
}
```

### Additional Info

Maybe also include logic to detect unnecessary `items instanceof Array`?

**Comment by kirkwaiblinger** [comment]
Hi @ulrichstark!

I think that this requests falls within scope for the `checkTypePredicates` option. Currently, that option is rather conservative about what it flags as unnecessary but we've discussed in https://github.com/typescript-eslint/typescript-eslint/issues/10997#issuecomment-3400495465 expanding it. The `Array.isArray()` could be seen as a special case of flagging code where the asserted operand is assignable to the asserted type.

So, I'd be in favor of going forward with expanding the `checkTypePredicates` option to include assignability checks, and testing the `Array.isArray()` behavior with that option. 

@ronami

**Comment by kirkwaiblinger** [comment]
Fixed by https://github.com/typescript-eslint/typescript-eslint/pull/12147 🥳
